### PR TITLE
fix: use NextJS default env local file name

### PR DESCRIPTION
**Pull request by DonghyukJacobJang**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Code style update
- [ ] Refactor (refactoring or adding test which isn't a fix or add a feature)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Did you test your solution?**

- [ ] I lightly tested it in one browser
- [x] I deeply tested it in several browsers
- [ ] I wrote tests around it (unit tests, integration tests, E2E tests)

## Problem Description

Using `BUILD_ENV` key and `.env.development` approach was introduced a long time ago before NextJS 9.4 release. 

## Solution Description

NextJS 9.4 + supports `.env.local` by default for the use of local environment. As a result, removing `BUILD_ENV` and renaming the file name.

reference: https://nextjs.org/docs/basic-features/environment-variables#loading-environment-variables

## Side Effects, Risks, Impact

- [x] N/A

**Aditional comments:**

### CODE REVIEW

**Comment by neo** [comment]
based on [this](https://nextjs.org/docs/basic-features/environment-variables#default-environment-variables), I think we might still want to use `.development` as the committed one? However, maybe we indeed don't need to specify `BUILD_ENV=development` in the dev npm script since it's the default

image

**Comment by neo** [comment]
also on a kind of related note, I was thinking that maybe we want to bring back `npm start` rather than `npm run dev`? I personally still mistakenly run `npm start` from time to time

**Comment by namniak** [comment]
> also on a kind of related note, I was thinking that maybe we want to bring back `npm start` rather than `npm run dev`? I personally still mistakenly run `npm start` from time to time

@neo I guess `npm start` would be valid rather for starting a prod node server (SSR) than local dev imho

**Comment by neo** [comment]
then I'd just need to try harder getting used to it 😂  was just considering how infrequent we need to run that prod SSR server by hand...

**Comment by edgardz** [comment]
I like the idea of having only these 2 envs:

.env.production (leveraging build time CI vars). Example:
```
NEXT_PUBLIC_ENVIRONMENT=$CI_ENV
```

.env.development (hard coding the CI vars we don't have while developing). Example:
```
NEXT_PUBLIC_ENVIRONMENT=local
```

This way we are able to do some cleanups:

- Remove this piece of logic from next.config.js:
```
require('dotenv').config({
  path: path.resolve(process.cwd(), `.env.${process.env.CI_ENV || process.env.NODE_ENV}`)
});
```

- Replacing the build scripts on package.json from
```
    "build:prod:static": "cross-env CI_ENV=prod next build && next export && cross-env CI_ENV=prod npm-run-all robots-txt-generator sitemap-generator",
    "build:staging:static": "cross-env CI_ENV=staging next build && next export && cross-env CI_ENV=prod npm-run-all sitemap-generator",
    "build:develop:static": "cross-env CI_ENV=develop next build && next export && cross-env CI_ENV=prod npm-run-all sitemap-generator",
```
to a single build script (bear in mind CI_ENV should be set on codeship-services.yml):
```
"build:static": "next build && next export && npm-run-all robots-txt-generator sitemap-generator",
```


It will also allow us to do conditional things in our code as well, like:
```
// hide GSAP warnings only in production, show on local, dev, staging...
gsap.config({ nullTargetWarn: process.env.NEXT_PUBLIC_ENVIRONMENT !== 'production' });
```

File: .codeship/artifact.sh [hunk 1]
```diff
@@ -2,7 +2,9 @@
 set -e
 
 #### Running build ####
-npm run build:prod:static
+export COMMIT_ID=$(git rev-parse --short HEAD)
```
Comment (edgardz) [inline .codeship/artifact.sh:5]: We can also count the number of commits to have a numeric representation of the current version:
```
export COMMIT_NUMBER=$(git rev-list --no-merges --count HEAD)
```

File: .env.production [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# reserved env key
+BUNDLE_ANALYZE=false
+
+# custom env key
+NEXT_PUBLIC_WEBSITE_SITE_URL=$WEBSITE_SITE_URL
+NEXT_PUBLIC_ENVIRONMENT=$CI_ENV
+NEXT_PUBLIC_BUILD_ID=$COMMIT_ID
+NEXT_PUBLIC_GTM_ID=$GTM_ID
```
Comment (edgardz) [inline .env.production:8]: Following my previous comment:
```
NEXT_PUBLIC_BUILD_NUMBER=$COMMIT_NUMBER
```

File: .codeship/artifact.sh (hunk 1 above)
Comment (DonghyukJacobJang) [reply .codeship/artifact.sh:5]: I am looking for a better versioning way with major and minor numbers. In my opinion, knowing the exact commit id upfront would be better to keep track for now.

Comment (edgardz) [reply .codeship/artifact.sh:5]: I see. Considering the boilerplate itself, it makes sense to have proper versioning.
But for everyday projects, most of the time we just need to be able to compare if staging is up to date with development for QA purposes, so having a simple commit number in the console, for instance, would do the job.

**Review by edgardz — APPROVED** [review]

**Comment by DonghyukJacobJang** [comment]
#126

**Pull request merged**